### PR TITLE
Fixed compiler warnings related to implicit float/double casts

### PR DESCRIPTION
**Pull request by sethalves**
Also removed a bunch of trailing whitespace and ctrl-Ms.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 508ae276c677e6cea69ead43af6b7f84460001c8 available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ad25a23860392b2b70043ccd286efd064da9532 available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Comment by ZappoMan** [comment]
are you sure about this? this feels wrong... lots more casts to doubles for things that really only need to be floats. Is this just about warnings? Does it improve performance?

**Comment by sethalves** [comment]
The casts to doubles are because printf and pals only take doubles (not floats).  Does seem to build anyway, so maybe there's no problem!

**Comment by ZappoMan** [comment]
Yeah - I just was reading up on that... apparently (x)printf doesn't have a format specifier for single precision float, and so they always get promoted to doubles... so retract my complaint.

But if you're really into cleaning up warnings, what about all the hidden virtual functions in the physics stuff. :)

**Comment by sethalves** [comment]
I don't see those warnings!  We need more compiler flags!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a841cc0b888144529952cc1877ee4ef755d5746 available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Comment by sethalves** [comment]
Do anyone attempting to review this:  remember to add ?w=1 to the end of the url.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7391aebf60e62644799836a3e82df5734f5967b1 available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0ec7d3ded82801439cabb6bc918a86e6d021b0ae available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

File: assignment-client/src/octree/OctreeServer.cpp [hunk 1]
```diff
@@ -610,13 +613,13 @@ bool OctreeServer::handleHTTPRequest(HTTPConnection* connection, const QUrl& url
...
-            ((float)totalBytesOfOctalCodes / (float)totalOutboundBytes) * AS_PERCENT);
+            (double)((totalBytesOfOctalCodes / (float)totalOutboundBytes) * AS_PERCENT));
         statsString += QString().sprintf("             Total BitMasks Bytes: %s bytes (%5.2f%%)\r\n",
             locale.toString((uint)totalBytesOfBitMasks).rightJustified(COLUMN_WIDTH, ' ').toLocal8Bit().constData(),
-            ((float)totalBytesOfBitMasks / (float)totalOutboundBytes) * AS_PERCENT);
+            (double)((totalBytesOfBitMasks / (float)totalOutboundBytes) * AS_PERCENT));
         statsString += QString().sprintf("                Total Color Bytes: %s bytes (%5.2f%%)\r\n",
             locale.toString((uint)totalBytesOfColor).rightJustified(COLUMN_WIDTH, ' ').toLocal8Bit().constData(),
-            ((float)totalBytesOfColor / (float)totalOutboundBytes) * AS_PERCENT);
+            (double)((totalBytesOfColor / (float)totalOutboundBytes) * AS_PERCENT));
```
Comment (AndrewMeadows) [inline assignment-client/src/octree/OctreeServer.cpp]: The original **float** cast was for **totalBytesOfColor** which is **quint64**, so now there is an implicit cast.

File: assignment-client/src/octree/OctreeServer.cpp [hunk 2]
```diff
@@ -726,7 +731,7 @@ bool OctreeServer::handleHTTPRequest(HTTPConnection* connection, const QUrl& url
             checkSum += OctreeElement::getChildrenCount(i);
             statsString += QString().sprintf("    Nodes with %d children:      %s nodes (%5.2f%%)\r\n", i,
                 locale.toString((uint)OctreeElement::getChildrenCount(i)).rightJustified(16, ' ').toLocal8Bit().constData(),
-                ((float)OctreeElement::getChildrenCount(i) / (float)nodeCount) * AS_PERCENT);
+                (double)((OctreeElement::getChildrenCount(i) / (float)nodeCount) * AS_PERCENT));
```
Comment (AndrewMeadows) [inline assignment-client/src/octree/OctreeServer.cpp]: Similarly here an implicit cast has been added.

**Comment by AndrewMeadows** [comment]
Net reduction of implicit casts is positive.  I'll merge this later today after demos.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53b5c8e6e7546a684aad34f271d253f72547b079 available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 021dc64789176fa5925ad9c14531877e01e565fa available at: https://gist.github.com/49302c24ffad9c6c06bd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5140.dmg

**Pull request merged**